packages/components/src/components/primitives/Tag/Tag.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC } from 'react';
+import {
+  Tag as ChakraTag,
+  TagLeftIcon,
+  TagRightIcon,
+  TagLabel,
+  useStyleConfig,
+} from '@chakra-ui/react';
+import { TagProps } from './types';
+
+export const Tag: FC<TagProps> = ({
+  label,
+  icon,
+  iconPosition,
+  size,
+  ...rest
+}) => {
+  const {
+    container,
+    label: labelStyle,
+    leftIcon,
+    rightIcon,
+  } = useStyleConfig('Tag', { size }) as any;
```

Comment:
Should this be `as StyleProps`? I think we are doing something similar for the `Avatar` component.

Reply:
type defined